### DISCUSSION TITLE
Usage of the prebuild images and existing workflow

### DISCUSSION THREAD

**Discussion by blublinsky** [elyra-ai/elyra#1724] [Q&A]
Thanks for creating of an amazing tool.

The question that I have is mostly about reuse. I went through the majority examples and it looks great. The things that I have not found:
1. If I have ready made docker image, can I include it in Elyra directly,as is without adding additional code?
2. If I have an existing pipeline, can I include it in Elyra as a sub workflow?

I know I can do it directly in Kubeflow pipelines, but it would be nice to do it in the editor

**Comment by lresende** [comment]
> 1. If I have ready made docker image, can I include it in Elyra directly, as is without adding additional code?

Yes, our examples are based on pre-existing community images as described in our [runtime-image docs](https://elyra.readthedocs.io/en/stable/user_guide/runtime-image-conf.html). I believe the only limitation is that the image should not rely on starting services from entrypoints/cmd as we will override it with our bootstrapper.


> 2.If I have an existing pipeline, can I include it in Elyra as a sub workflow?

At this moment this is not available as a feature in the visual editor. We would actually need an enhancement on the actual Elyra Canvas to support that, and we will have to implement that support when it gets available from the canvas component. Having said that, if your existing pipeline is just a script file (e.g. using DSL) then you could try to include that as a node on this pipeline, but dependency management might be an issue (e.g. if the next node needs to wait for the pipeline completion before it can continue execution) but we can probably find some workaround for that.

**Reply by ptitzler** [reply]
> I believe the only limitation is that the image should not rely on starting services from entrypoints/cmd as we will override it with our bootstrapper.

See https://elyra.readthedocs.io/en/stable/recipes/creating-a-custom-runtime-image.html
